### PR TITLE
Add Re-Import and tests

### PR DESCRIPTION
**Pull request by wurstbrot**
Work in Progress

### CODE REVIEW

**Comment by wurstbrot** [comment]
Please merge

File: scb-persistenceproviders/defectdojo-persistenceprovider/src/main/java/io/securecodebox/persistence/DefectDojoService.java [hunk 1]
```diff
@@ -61,12 +62,16 @@
...
 
     protected static final String DATE_FORMAT = "yyyy-MM-dd";
+    protected static final String DATE_TIME_FORMAT = "yyyy-MM-dd hh:m:ss";
 
     Clock clock = Clock.systemDefaultZone();
 
     private String currentDate() {
         return LocalDate.now(clock).format(DateTimeFormatter.ofPattern(DATE_FORMAT));
     }
+    private String currentDateTime() {
```
Comment (J12934) [inline scb-persistenceproviders/defectdojo-persistenceprovider/src/main/java/io/securecodebox/persistence/DefectDojoService.java]: Does DefectDojo now support more detailed dates (with hour and minutes) for the tests?
If i remember correctly we tried to add them like this but it didn't used to work.

File: scb-persistenceproviders/defectdojo-persistenceprovider/src/main/java/io/securecodebox/persistence/DefectDojoService.java [hunk 2]
```diff
@@ -314,21 +442,24 @@ public void deleteUnusedBranches(List<String> existingBranches, String producNam
...
+        if(existingBranches == null) {
+            LOG.error("No existing branches given, this will lead to nullpointer");
+        }
         RestTemplate restTemplate = new RestTemplate();
         
         //get existing branches
         List<EngagementResponse> engagementPayloads = getEngagementsForProduct(productId, 0);
         for(EngagementResponse engagementPayload : engagementPayloads) {
-            boolean branchExists = false;
+            boolean branchExists = false
```
Comment (J12934) [inline scb-persistenceproviders/defectdojo-persistenceprovider/src/main/java/io/securecodebox/persistence/DefectDojoService.java]: Missing semicolon is breaking the build.

```suggestion
            boolean branchExists = false;
```

**Comment by J12934** [comment]
The existing persistence provider functionality works unhindered.
Will merge once the build is green again. 👍

File: scb-persistenceproviders/defectdojo-persistenceprovider/src/main/java/io/securecodebox/persistence/DefectDojoService.java (hunk 1 above)
Comment (wurstbrot) [reply scb-persistenceproviders/defectdojo-persistenceprovider/src/main/java/io/securecodebox/persistence/DefectDojoService.java]: No, only tests needs to be defined with time. As you mostly create it with an engagement, you didn't had contact so far.

**Pull request merged**